### PR TITLE
[PM-5154] Implement combined view for passwords and passkeys on iOS Autofill extension

### PR DESCRIPTION
**Pull request by fedemkr**
## Type of change
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Implement combined view for passwords and passkeys on iOS Autofill extension with in-view search feature.

## Code changes

* **CredentialProviderViewController[Passkeys]:** Moved `PrepareCredentialList` to Passkeys file and reuse the one without passkeys request parameters from this one saving the parameters in the context.
* **LoginListViewController:**
  * Added Loading indicator
  * Implemented different names for the title depending on the context
  * Added Search feature inside this list, the toolbar search button toggles the visibility of the search bar
  * Implemented Fido2 `GetAssertion` for the logic for the combined passwords and passkeys
  * Improved empty state
  * Improved loading and reloading items
* **BaseLoginListTableSource:** Main point for implementing the logic of cipher loading, taking into consideration the possibility of splitting passwords and passkeys into two separate sections and the specific loading and filling of them from the context.
* **ExtensionTableSource:** Improved this base class to be extendable and have the new UI and passkeys logic into account.
* **Fido2GetAssertionFromListUserInterface:** New class to have the specific logic to choose a passkey from the combined view of Passwords + Passkeys.
* **Fido2AuthenticatorService:** Added logging when an unknown exception is raised in the process
* **LoginSearchViewController:** Fixed issue on loading items
* **MainInterface.storyboard:** Added search bar and loading indicator on Logins scene and improved constraints
* **Context:** Added passkeys request parameters and task completion source for picking a credential from list combined flow and some helper flags
* **AutofillHelpers:** Simplified the method to select a row for Password

### Auto-fill combined view: Passwords + Passkeys
Auto-fill combined view

### Auto-fill combined view with search
Auto-fill combined view with search

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/iOS.Autofill/LoginListViewController.cs [hunk 1]
```diff
@@ -306,62 +470,53 @@ protected override void Dispose(bool disposing)
...
-                    }
-
-                    return new UIView(CGRect.Empty);
-                }
-                catch (Exception ex)
-                {
-                    LoggerHelper.LogEvenIfCantBeResolved(ex);
-                    return new UIView();
-                }
+                _isLoading = false;
```
Comment (andrebispo5) [inline src/iOS.Autofill/LoginListViewController.cs:497]: ❓ If there is an exception we are currently only defaulting the `_isLoading`, should we also change the `Hidden` property of `TableView`, `_searchBar` and `_loadingView`?

File: src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs [hunk 2]
```diff
@@ -31,6 +43,207 @@ public BaseLoginListTableSource(T controller)
...
+
+                if (!_allowedFido2CipherIds.Any()
+                    ||
+                    _allowedFido2CipherIds.Contains(item.Id))
+                {
+                    fido2CiphersToInsert.Add(item.ToPasskeyListItemCipherViewModel());
+                }
+            }
+
+            if (fido2CiphersToInsert.Any())
```
Comment (andrebispo5) [inline src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs:82]: :pick: Invert to reduce nesting

File: src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs [hunk 3]
```diff
@@ -31,6 +43,207 @@ public BaseLoginListTableSource(T controller)
...
+                LoadFido2Ciphers(ct);
+            }
+
+            Controller.OnItemsLoaded(searchFilter);
+        }
+
+        private void LoadFido2Ciphers(CancellationToken ct)
+        {
+            var fido2CiphersToInsert = new List<CipherViewModel>();
+            foreach (var item in Items.Where(i => i?.CipherView?.HasFido2Credential == true))
```
Comment (andrebispo5) [inline src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs:70]: ❓ Noticed that couple lines down below you were null checking `Items` should it be done here as well?

File: src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs [hunk 4]
```diff
@@ -80,6 +307,31 @@ private async Task SelectRowForPasskeyCreationAsync(UITableView tableView, NSInd
 
             Context.PickCredentialForFido2CreationTcs.SetResult((item.Id, null));
         }
+
+        private async Task<CipherViewModel> DeselectRowAndGetItemAsync(UITableView tableView, NSIndexPath indexPath)
+        {
+            tableView.DeselectRow(indexPath, true);
+            tableView.EndEditing(true);
+
+            var item = Items.ElementAtOrDefault(GetIndexForItemAt(tableView, indexPath));
```
Comment (andrebispo5) [inline src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs:316]: ❓ Items null check?

File: src/iOS.Autofill/Utilities/Fido2GetAssertionFromListUserInterface.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            _hasVaultBeenUnlockedInThisTransaction = hasVaultBeenUnlockedInThisTransaction;
+            _verifyUserCallback = verifyUserCallback;
+            _onAllowedFido2Credentials = onAllowedFido2Credentials;
+        }
+
+        public bool HasVaultBeenUnlockedInThisTransaction { get; private set; }
+
+        public async Task<(string CipherId, bool UserVerified)> PickCredentialAsync(Fido2GetAssertionUserInterfaceCredential[] credentials)
+        {
+            if (credentials.Length == 0)
```
Comment (andrebispo5) [inline src/iOS.Autofill/Utilities/Fido2GetAssertionFromListUserInterface.cs:45]: Null check `credentials`

File: src/iOS.Autofill/Utilities/Fido2GetAssertionFromListUserInterface.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            {
+                throw new NotAllowedError();
+            }
+
+            HasVaultBeenUnlockedInThisTransaction = _hasVaultBeenUnlockedInThisTransaction();
+
+            _onAllowedFido2Credentials?.Invoke(credentials?.Select(c => c.CipherId).ToList() ?? new List<string>());
```
Comment (andrebispo5) [inline src/iOS.Autofill/Utilities/Fido2GetAssertionFromListUserInterface.cs:52]: `credentials` null check can be removed here if done above.

**Review by andrebispo5 — CHANGES_REQUESTED** [review]
Looking good, just a couple of small things. 👍

File: src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs (hunk 4 above)
Comment (fedemkr) [reply src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs:316]: If `null` it's ok to break here so we know something is wrong.

File: src/iOS.Autofill/LoginListViewController.cs (hunk 1 above)
Comment (fedemkr) [reply src/iOS.Autofill/LoginListViewController.cs:497]: Hmmm, if there's an exception there then the view would be in a corrupted state, so I wouldn't "un-hide" the other views. I'll add an alert to the user with a general error message; thanks for bringing this up.

File: src/iOS.Autofill/Utilities/Fido2GetAssertionFromListUserInterface.cs (hunk 5 above)
Comment (fedemkr) [reply src/iOS.Autofill/Utilities/Fido2GetAssertionFromListUserInterface.cs:45]: It's never `null` and if it's it should break.

File: src/iOS.Autofill/Utilities/Fido2GetAssertionFromListUserInterface.cs (hunk 6 above)
Comment (fedemkr) [reply src/iOS.Autofill/Utilities/Fido2GetAssertionFromListUserInterface.cs:52]: Nice catch!

**Review by andrebispo5 — APPROVED** [review]
Perfect 🎉

**Pull request merged**